src/dhcpv6.c

```diff
@@ -0,0 +1,342 @@
+    /* sanity check */
+    if (d1 == NULL) {
+        return;
+    }
+
+    /* print out the data we want from the messages */
+    if (d1->message_count) {
+        zprintf(f, ",\"dhcpv6\":[");
+        for (i = 0; i < d1->message_count; i++) {
+            const char *disp_buffer = NULL;
+            const dhcp_v6_message_t *msg = &d1->messages[i];
+            uint8_t *ptr = (uint8_t*)msg->data;
+
+            /* pesky JSON commas to seperate elements */
+            if (!first_time) {
+                zprintf(f, ",");
+            } else {
+                first_time = 0;
+            }
+
+            /* printout the message type and transaction id */
+            disp_buffer = dhcpv6_msg_to_string(msg->msg_type);
+            if (disp_buffer == NULL) {
+                zprintf(f, "{\"type\":\"%u\",",msg->msg_type);
+            } else {
+                zprintf(f, "{\"type\":\"%s\",",disp_buffer);
+            }
+            zprintf(f, "\"transid\":\"%x\",",msg->trans_id);
+
+            /* first 4 bytes of the option are the type and length */
+            dhcpv6_option = *ptr << 8 | *(ptr+1);
+            ptr += 2;
+            dhcpv6_opt_len = *ptr << 8 | *(ptr+1);
+            ptr += 2;
+
+            /* printout the option and length */
+            disp_buffer = dhcpv6_option_to_string(dhcpv6_option);
+            if (disp_buffer == NULL) {
+                zprintf(f, "\"option\":\"%u\",",dhcpv6_option);
+            } else {
+                zprintf(f, "\"option\":\"%s\",",dhcpv6_option_to_string(dhcpv6_option));
+            }
+
+#ifdef DHCPV6_DEBUG
+            zprintf(f, "\"optlen\":%u,",dhcpv6_opt_len);
+            if ((dhcpv6_option == DHCPV6_CLIENTID) || (dhcpv6_option == DHCPV6_SERVERID)) {
+
+                zprintf(f, "\"macaddr\":\"%.2x:%.2x:%.2x:%.2x:%.2x:%.2x\",",
+                            *(ptr+8),*(ptr+9),*(ptr+10),
+                            *(ptr+11),*(ptr+12),*(ptr+13));
+            }
+
+            /* show the remaning bytes of the message */
+            int k = 0;
+            int max_bytes = 0;
+            max_bytes = (dhcpv6_opt_len < MAX_DHCP_V6_MSG_LEN) ? dhcpv6_opt_len : MAX_DHCP_V6_MSG_LEN;
+            zprintf(f, "\"data\":\"");
+            for (k=0; k < max_bytes; ++k) {
+                zprintf(f, "%.2x", *(ptr+k));
+            }
+            zprintf(f, "\"}");
+        }
+#else
+            /* for CLIENT ID and SERVER ID, printout the mac address */
+            if ((dhcpv6_option == DHCPV6_CLIENTID) || (dhcpv6_option == DHCPV6_SERVERID)) {
+                zprintf(f, "\"optlen\":%u,",dhcpv6_opt_len);
+                zprintf(f, "\"macaddr\":\"%.2x:%.2x:%.2x:%.2x:%.2x:%.2x\"}",
+                            *(ptr+8),*(ptr+9),*(ptr+10),
+                            *(ptr+11),*(ptr+12),*(ptr+13));
+            } else {
+                zprintf(f, "\"optlen\":%u}",dhcpv6_opt_len);
+            }
+        }
+#endif
+        zprintf(f, "]");
+    }
+
+    /* sanity check and for compiler complain */
```

Comment:
You can use __attribute__ ((unused)) const dhcpv6_t *d2 at function declaration and you won't need this block.

Reply:
not sure how widely supported that attribute is across various compilers. This is safer and will likely get optimized out by the compiler since the true condition result equates to the false condition.